source/configuration_manual/stats/index.rst

```diff
@@ -52,6 +52,21 @@ generation based on event fields' values.  Each field listed in the
 generated metrics are indistinguishable from those statically defined
 in the config file.
 
+Dovecot supports a number of aggregate functions that can be used to
```

Comment:
Are multiple group_by statements possible within the same block?

Reply:
I'm not sure what exactly you are asking, so I'll answer both cases...

*multiple `group_by` lines:* The settings parser doesn't prohibit it, but IIRC it uses the last one's value.

*nested quantization:* Yes, this is supported by the code (but I'm not sure if we'll be able to stuff it into openmetrics format). For example, the config:

```
metric test4 {
        event_name = imap_command_finished
        fields = bytes_out bytes_in
        group_by = cmd_name duration:exponential:1:10:10 bytes_out:linear:0:1000:100
}
```
produces:

```
$ doveadm stats dump
test4	duration	7	111654	69	78186	15950.57	370	27665.17	78186
test4	bytes_out	7	1284	44	587	183.43	79	191.87	587
test4	bytes_in	7	40	0	11	5.71	8	3.95	11
test4_select	duration	1	78186	78186	78186	78186.00	78186	0.00	78186
test4_select	bytes_out	1	352	352	352	352.00	352	0.00	352
test4_select	bytes_in	1	5	5	5	5.00	5	0.00	5
test4_select_duration_10001_100000	duration	1	78186	78186	78186	78186.00	78186	0.00	78186
test4_select_duration_10001_100000	bytes_out	1	352	352	352	352.00	352	0.00	352
test4_select_duration_10001_100000	bytes_in	1	5	5	5	5.00	5	0.00	5
test4_select_duration_10001_100000_bytes_out_301_400	duration	1	78186	78186	78186	78186.00	78186	0.00	78186
test4_select_duration_10001_100000_bytes_out_301_400	bytes_out	1	352	352	352	352.00	352	0.00	352
test4_select_duration_10001_100000_bytes_out_301_400	bytes_in	1	5	5	5	5.00	5	0.00	5
test4_fetch	duration	4	33197	313	32014	8299.25	435	13691.89	32014
test4_fetch	bytes_out	4	824	78	587	206.00	79	219.97	587
test4_fetch	bytes_in	4	35	8	11	8.75	8	1.30	11
test4_fetch_duration_10001_100000	duration	1	32014	32014	32014	32014.00	32014	0.00	32014
test4_fetch_duration_10001_100000	bytes_out	1	78	78	78	78.00	78	0.00	78
test4_fetch_duration_10001_100000	bytes_in	1	8	8	8	8.00	8	0.00	8
test4_fetch_duration_10001_100000_bytes_out_1_100	duration	1	32014	32014	32014	32014.00	32014	0.00	32014
test4_fetch_duration_10001_100000_bytes_out_1_100	bytes_out	1	78	78	78	78.00	78	0.00	78
test4_fetch_duration_10001_100000_bytes_out_1_100	bytes_in	1	8	8	8	8.00	8	0.00	8
test4_fetch_duration_101_1000	duration	3	1183	313	500	394.33	370	78.26	500
test4_fetch_duration_101_1000	bytes_out	3	746	79	587	248.67	80	239.24	587
test4_fetch_duration_101_1000	bytes_in	3	27	8	11	9.00	8	1.41	11
test4_fetch_duration_101_1000_bytes_out_1_100	duration	2	683	313	370	341.50	341	28.50	370
test4_fetch_duration_101_1000_bytes_out_1_100	bytes_out	2	159	79	80	79.50	790.50	80
test4_fetch_duration_101_1000_bytes_out_1_100	bytes_in	2	16	8	8	8.00	80.00	8
test4_fetch_duration_101_1000_bytes_out_501_600	duration	1	500	500	500	500.00	500	0.00	500
test4_fetch_duration_101_1000_bytes_out_501_600	bytes_out	1	587	587	587	587.00	587	0.00	587
test4_fetch_duration_101_1000_bytes_out_501_600	bytes_in	1	11	11	11	11.00	110.00	11
test4_close	duration	1	202	202	202	202.00	202	0.00	202
test4_close	bytes_out	1	44	44	44	44.00	44	0.00	44
test4_close	bytes_in	1	0	0	0	0.00	0	0.00	0
test4_close_duration_101_1000	duration	1	202	202	202	202.00	202	0.00	202
test4_close_duration_101_1000	bytes_out	1	44	44	44	44.00	44	0.00	44
test4_close_duration_101_1000	bytes_in	1	0	0	0	0.00	0	0.00	0
test4_close_duration_101_1000_bytes_out_1_100	duration	1	202	202	202	202.00	202	0.00	202
test4_close_duration_101_1000_bytes_out_1_100	bytes_out	1	44	44	44	44.00	440.00	44
test4_close_duration_101_1000_bytes_out_1_100	bytes_in	1	0	0	0	0.00	00.00	0
test4_logout	duration	1	69	69	69	69.00	69	0.00	69
test4_logout	bytes_out	1	64	64	64	64.00	64	0.00	64
test4_logout	bytes_in	1	0	0	0	0.00	0	0.00	0
test4_logout_duration_11_100	duration	1	69	69	69	69.00	69	0.00	69
test4_logout_duration_11_100	bytes_out	1	64	64	64	64.00	64	0.00	64
test4_logout_duration_11_100	bytes_in	1	0	0	0	0.00	0	0.00	0
test4_logout_duration_11_100_bytes_out_1_100	duration	1	69	69	69	69.00	690.00	69
test4_logout_duration_11_100_bytes_out_1_100	bytes_out	1	64	64	64	64.00	640.00	64
test4_logout_duration_11_100_bytes_out_1_100	bytes_in	1	0	0	0	0.00	00.00	0
```